src/filters/array.rs

```diff
@@ -0,0 +1,217 @@
+use liquid_value::Value;
+
+use super::{check_args_len, invalid_input};
+use interpreter::FilterResult;
+use std::fmt::Write;
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the given value appended at the end.
+pub fn push(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 1, 0)?;
+
+    let value = args[0].clone();
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+    array.push(value);
+
+    Ok(Value::array(array))
+}
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the last element removed.
+pub fn pop(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 0, 0)?;
+
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+    array.pop();
+
+    Ok(Value::array(array))
+}
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the given value appended at the beginning.
+pub fn unshift(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 1, 0)?;
+
+    let value = args[0].clone();
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+    array.insert(0, value);
+
+    Ok(Value::array(array))
+}
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the first element removed.
+pub fn shift(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 0, 0)?;
+
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+
+    if !array.is_empty() {
+        array.remove(0);
+    }
+
+    Ok(Value::array(array))
+}
+
+/// Convert an array into a sentence.
+pub fn array_to_sentence_string(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 0, 1)?;
+
+    let connector = args
+        .get(0)
+        .and_then(|arg| Some(arg.to_string()))
```

Comment:
Sorry I missed this earlier.  Does this need to be `to_string` or can it be `to_str`?

For example:
```rust
let possible_connector = args
    .get(0)
    .and_then(|arg| Some(arg.to_str()));
let connector = possible_connector
    .map(|s| s.as_ref)
    .unwrap_or_else("and");
```
- `possible_connector` would be a `Option<Cow<'static, str>>`
- `connector` would be a `&str`